### PR TITLE
Add intake and shooter subsystems

### PR DESCRIPTION
**Pull request by sanashah007**

### CODE REVIEW

File: src/main/java/org/team1540/robot2022/commands/shooter/Shooter.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    public void config_kD(double kD) {
+        shooterMotorFront.config_kD(0, kD);
+        shooterMotorRear.config_kD(0, kD);
+
+    }
+
+    public void setShooterMotorA(){
+
+    }
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/shooter/Shooter.java:130]:
```suggestion
```

File: src/main/java/org/team1540/robot2022/commands/hood/HoodSet.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package org.team1540.robot2022.commands.hood;
+
+;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class HoodSet extends CommandBase {
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/hood/HoodSet.java:6]: If we're going for consistency, should this be `HoodSetCommand` instead?

File: src/main/java/org/team1540/robot2022/commands/shooter/ShooterManualControl.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    
+    @Override
+    public void execute(){
+        System.out.println("deploying correctly"); 
+        if(copilot.getLeftBumper()){
+            shooter.setVelocityRPMA(0.3);
+        }
+        if(copilot.getRightBumper()){
+            shooter.setVelocityRPMB(0.3); 
+        }
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/shooter/ShooterManualControl.java:28]: Logic in here should also probably be moved into commands

File: src/main/java/org/team1540/robot2022/commands/shooter/ShooterTesting.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
+package org.team1540.robot2022.commands.shooter;
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/shooter/ShooterTesting.java:1]: What is this file used for?

File: src/main/java/org/team1540/robot2022/utils/InterpolationTable.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package org.team1540.robot2022;
+package org.team1540.robot2022.utils;
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/utils/InterpolationTable.java:1]: This is more like `Constants.java` or `RamseteConfig.java` than a util, I don't think it makes sense to move it.

**Review by rutmanz — CHANGES_REQUESTED** [review]

File: src/main/java/org/team1540/robot2022/commands/hood/HoodSetCommand.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package org.team1540.robot2022.commands.hood;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class HoodSetCommand extends CommandBase {
+    private final boolean state;
+    private final Hood hood;
+
+    public HoodSetCommand(Hood hood, Boolean state) {
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/hood/HoodSetCommand.java:9]:
```suggestion
    public HoodSetCommand(Hood hood, boolean state) {
```
I believe `Boolean` is different than `boolean`

File: src/main/java/org/team1540/robot2022/commands/intake/Intake.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    public Intake() {
+        Constants.IntakeConstants.currentLimitConfig.applyTo(motor);
+    }
+
+    @Override
+    public void periodic() {
+    }
+
+    public void setTilt(Boolean state) {
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/intake/Intake.java:23]:
```suggestion
    public void setFolded(boolean state) {
```
`setTilt` implies to me that it's not a binary state
`boolean` vs `Boolean` again (https://stackoverflow.com/questions/3728616/boolean-vs-boolean-in-java)

File: src/main/java/org/team1540/robot2022/commands/intake/IntakeSpinCommand.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    @Override
+    public void initialize() {
+        intake.setPercent(this.speed);
+    }
+
+    @Override
+    public void execute() {
+    }
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/intake/IntakeSpinCommand.java:21]: perhaps also add an `end()` method to turn off the motors when it's un-scheduled
```java
@Override
public void end(boolean isInterrupted) {
    intake.setPercent(0)
}
```

File: src/main/java/org/team1540/robot2022/commands/intake/IntakeTiltCommand.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
+package org.team1540.robot2022.commands.intake;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class IntakeTiltCommand extends CommandBase {
```
Comment (rutmanz) [inline src/main/java/org/team1540/robot2022/commands/intake/IntakeTiltCommand.java:5]: Same comment about tilt vs fold from before

File: src/main/java/org/team1540/robot2022/commands/intake/IntakeSpinCommand.java (hunk 8 above)
Comment (natesales) [reply src/main/java/org/team1540/robot2022/commands/intake/IntakeSpinCommand.java:21]: Nice catch!

Comment (rutmanz) [reply src/main/java/org/team1540/robot2022/commands/intake/IntakeSpinCommand.java:22]: That's in the `execute()` method, not `end(...)`

Comment (natesales) [reply src/main/java/org/team1540/robot2022/commands/intake/IntakeSpinCommand.java:21]: Doh!

**Review by rutmanz — APPROVED** [review]

**Review by natesales — APPROVED** [review]

**Pull request merged**